lib/expand/join.go

```diff
@@ -1053,16 +1076,16 @@ func (p *Peer) emitAuditEvent(ctx operationContext) error {
 	return nil
 }
 
-func (p *Peer) operationContext() (*operationContext, error) {
+func (p *Peer) operationContext(ctx context.Context) (*operationContext, error) {
```

Comment:
That's a bug that prevented the agent to leave the cluster when aborted as it would receive on `p.ctx.Done()` immediately and exit with an error before.